File: backend/Scheduler.Models/Schedule.cs
namespace Scheduler.Models;

public class Schedule
{
    public int Id { get; set; }

    public string? Name { get; set; }

    public string SrcPath { get; set; } = null!;

    public string DestPath { get; set; } = null!;
    // NOTE: because `nullable` is enabled, in the default constructor we are not providing any 
    // value to the "non-nullable" properties. so the compiler will give warnings/error for it.
    // so either we need use `required` modifier or `null!` to fix.

    public required string CronExpression { get; set; }
    public bool Enabled { get; set; }
}

public class BackupRun
{
    public int Id { get; set; }

    public int ScheduleId { get; set; }

    public int BackupRunType { get; set; }

    public int Status { get; set; }

    public DateTime? StartTime { get; set; }

    public DateTime? CompletedAt { get; set; }

    public int? ExitCode { get; set; }

    public string? Logs { get; set; }
}

File: backend/samay-worker-go/models/schedule.go
package models

import "time"

type Schedule struct {
	ID             int     `db:"id" json:"id"`
	Name           *string `db:"name" json:"name,omitempty"` // nullable in DB
	SrcPath        string  `db:"srcpath" json:"srcPath"`
	DestPath       string  `db:"destpath" json:"destPath"`
	CronExpression string  `db:"cronexpression" json:"cronExpression"`
	Enabled        bool    `db:"enabled" json:"enabled"`
}

type BackupRun struct {
	Id            int        `db:"id" json:"id"`
	ScheduleId    *int       `db:"scheduleid" json:"scheduleId"`
	BackupRunType int        `db:"backupruntype" json:"backupRunType"`
	Status        int        `db:"status" json:"status"`
	StartTime     time.Time  `db:"starttime" json:"startTime"`
	CompletedAt   *time.Time `db:"completedat" json:"completedAt"`
	ExitCode      *int       `db:"exitcode" json:"exitCode"`
	Logs          *string    `db:"logs" json:"logs"`
}
